spinoso-string/src/enc/utf8/mod.rs

```diff
@@ -1082,4 +1191,35 @@ mod tests {
         let s = Utf8String::from(b"\xF0\x9F\x87");
         assert_eq!(s.chr(), b"\xF0");
     }
+
+    #[test]
+    fn get_char_slice_valid_range() {
+        let s = Utf8String::from(b"a\xF0\x9F\x92\x8E\xFF".to_vec()); // "a💎\xFF"
+        assert_eq!(s.get_char_slice(Range { start: 0, end: 0 }), Some(&b""[..]));
+        assert_eq!(s.get_char_slice(Range { start: 0, end: 1 }), Some(&b"a"[..]));
+        assert_eq!(s.get_char_slice(Range { start: 0, end: 2 }), Some("a💎".as_bytes()));
+        assert_eq!(
+            s.get_char_slice(Range { start: 0, end: 3 }),
+            Some(&b"a\xF0\x9F\x92\x8E\xFF"[..])
+        );
+        assert_eq!(
+            s.get_char_slice(Range { start: 0, end: 4 }),
+            Some(&b"a\xF0\x9F\x92\x8E\xFF"[..])
+        );
+        assert_eq!(s.get_char_slice(Range { start: 1, end: 1 }), Some(&b""[..]));
+        assert_eq!(s.get_char_slice(Range { start: 1, end: 2 }), Some("💎".as_bytes()));
+        assert_eq!(
+            s.get_char_slice(Range { start: 1, end: 3 }),
+            Some(&b"\xF0\x9F\x92\x8E\xFF"[..])
+        );
+    }
+
+    #[test]
+    fn get_char_slice_invalid_range() {
+        let s = Utf8String::from(b"a\xF0\x9F\x92\x8E\xFF".to_vec()); // "a💎\xFF"
+        assert_eq!(s.get_char_slice(Range { start: 4, end: 5 }), None);
```

Comment:
can we add the following test cases to ensure we have branch coverage on all parts of the fast path?

- `Range { start: 7, end: 10 }`
- `Range { start: 10, end: 8 }`
- `Range { start: 10, end: 5 }`
- `Range { start: 10, end: 2 }`

Reply:
Added in d25a1c2